source/ui/beamline/AMXYThetaControlMoveButton.cpp

```diff
@@ -0,0 +1,282 @@
+													   AMControl* controlX,
+													   AMControl* controlY,
+													   AMControl* controlTheta,
+													   QList<double> stepSizes,
+													   bool directionReversed,
+													   bool direction2Reversed,
+													   double angleOffset) : QToolButton(parent)
+{
+	directionXReversed_ = directionReversed;
+	directionYReversed_ = direction2Reversed;
+
+	angleOffset_ = angleOffset;
+
+	controlX_ = 0;
+	setControlX(controlX);
+
+	controlY_ = 0;
+	setControlY(controlY);
+
+	controlTheta_ = 0;
+	setControlTheta(controlTheta);
+
+	checkStatus();
+
+	setToolTip("XYTheta Move");
+
+	stepSizes_ = stepSizes;
+	if(stepSizes_.isEmpty())
+		stepSizes_ << 1.0;	// just to make sure we always have at least one, so that stepSizeIndex_ is valid.
+	stepSizeIndex_ = 0;
+
+	contextMenu_ = 0;
+	setContextMenuPolicy(Qt::CustomContextMenu);
+	connect(this, SIGNAL(customContextMenuRequested(QPoint)), this, SLOT(onCustomContextMenuRequested(QPoint)));
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+	connect(this, SIGNAL(clicked()), this, SLOT(onButtonClicked()));
+
+}
+
+
+
+
+void AMXYThetaControlMoveButton::setControlX(AMControl *control)
+{
+	if(control == controlX_)
+		return;
+
+	if(contextMenu_) {
+		contextMenu_->deleteLater();
+		contextMenu_ = 0;
+	}
+
+	if(controlX_) {
+		disconnect(controlX_, 0, this, 0);
+	}
+
+	controlX_ = control;
+	if(controlX_) {
+		connect(controlX_, SIGNAL(destroyed()), this, SLOT(onControlXDestroyed()));
+		connect(controlX_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMXYThetaControlMoveButton::setControlY(AMControl *control2)
+{
+	if(control2 == controlY_)
+		return;
+
+	if(contextMenu_) {
+		contextMenu_->deleteLater();
+		contextMenu_ = 0;
+	}
+
+	if(controlY_) {
+		disconnect(controlY_, 0, this, 0);
+	}
+
+	controlY_ = control2;
+	if(controlX_) {
+		connect(controlY_, SIGNAL(destroyed()), this, SLOT(onControlYDestroyed()));
+		connect(controlY_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMXYThetaControlMoveButton::setControlTheta(AMControl *controlR)
+{
+	if(controlR == controlTheta_)
+		return;
+
+	if(contextMenu_) {
+		contextMenu_->deleteLater();
+		contextMenu_ = 0;
+	}
+
+	if(controlTheta_) {
+		disconnect(controlTheta_, 0, this, 0);
+	}
+
+	controlTheta_ = controlR;
+	if(controlX_) {
+		connect(controlTheta_, SIGNAL(destroyed()), this, SLOT(onControlThetaDestroyed()));
+		connect(controlTheta_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
```

Comment:
Weak.  I'll fix these.  The buttons don't even display these text strings.  I'll probably just remove this.  However the check before connection could actually matter.  Too much faith in search and replace I guess.  Uncool.